### PR TITLE
Idempotent rerender

### PR DESCRIPTION
**Pull request by mmun**

### CODE REVIEW

**Comment by rwjblue** [comment]
Published as v0.12.0

**Comment by rondale-sc** [comment]
:+1:

**Comment by matthewrobb** [comment]
Just curious if there are plans to release maybe a blog post or something covering the meat of this PR? It's very difficult to tell exactly what WAS done between this PR and the reactive rerender PR.

**Comment by jonnii** [comment]
@matthewrobb the LIFECYCLE.md file contains a good description of the changes.

File: LIFECYCLE.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+  },
+
+  isStable: function(state, env, scope, params, hash) {
+    // This function is invoked after the first render; it checks to see
+    // whether the node is "stable". If the node is unstable, its
+    // existing content will be removed and the `render` function is
+    // called again to produce new values.
+  },
+
+  rerender: function(morph, env, scope, params, hash, template, inverse
```
Comment (gmile) [inline LIFECYCLE.md]: Missing trailing coma?

**Pull request merged**